src/azul/service/responseobjects/hca_response_v5.py

```diff
@@ -305,15 +321,20 @@ def _create_zipped_bdbag(participant_file_object: StringIO, sample_file_object:
             return bdbag_api.archive_bag(bag_path, 'zip')
 
     def _iterate_hit_tsv(self, es_search_hit, writer):
+        # if entity_type == 'files':
+        #     assert len(hit_dict['contents'][entity_type]) == 1
+
         hit_dict = es_search_hit.to_dict()
-        assert len(hit_dict['contents']['files']) == 1
-        file = hit_dict['contents']['files'][0]
-        file_fields = self._translate(file, 'contents.files')
 
-        for bundle in hit_dict['bundles']:
-            # FIXME: If a file is in multiple bundles, the manifest will list it twice. `hca dss download_manifest`
-            # would download the file twice (https://github.com/DataBiosphere/azul/issues/423).
-            writer.writerow(self._translate(bundle, 'bundles') + file_fields)
+        inner_entity_fields = []
+        for doc_path, column_mapping in self.manifest_entries.items():
+            doc_path = doc_path.split('.')
+            entities = hit_dict
+            for key in doc_path:
+                entities = entities.get(key, [{}])
```

Comment:
This works if the missing key is the last component of doc path. If earlier components were missing, like `contents` for example, the next loop iteration would attempt to do `[{}].get(key)` and raise an exception.